### PR TITLE
Nc games 4

### PR DESCRIPTION
**Pull request by sasjaWijdenes**
Added functionality for getting a review by its id along with error handling for non-existant references and invalid id's. Full TTD for both.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,14 +31,54 @@ describe("/api", () => {
...
         .get("/api/invalidRoute")
         .expect(404)
         .then(({ body }) => {
           expect(body.msg).toEqual("That route does not exist");
         });
     });
+    test("400: Should return when passed an id that is not valid", () => {
+      return request(app)
+        .get("/api/reviews/InvalidId")
+        .expect(404)
```
Comment (itskatherine) [inline __tests__/app.test.js:70]: Expecting a 404 instead of a 400?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -36,14 +31,54 @@ describe("/api", () => {
...
+              title: "Agricola",
+              designer: "Uwe Rosenberg",
+              owner: "mallionaire",
+              review_img_url:
+                "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+              review_body: "Farmyard fun!",
+              category: "euro game",
+              created_at: "2021-01-18T10:00:20.514Z",
+              votes: 1,
+              review_id: 1,
```
Comment (itskatherine) [inline __tests__/app.test.js:52]: Niiiiice test - testing for the specific article we're looking for 👍

**Review by itskatherine — COMMENTED** [review]
Great stuff Sasja! Good to merge once you've checked the comments 👍

File: __tests__/app.test.js (hunk 1 above)
Comment (sasjaWijdenes) [reply __tests__/app.test.js:70]: oooops lol, got them the wrong way round initialy and forgot to change it.

**Pull request merged**